ui/src/urn.test.ts

```diff
@@ -0,0 +1,78 @@
+/**
+ * The jsdom environment lacks `TextEncoder` required by `multibase`,
+ * see https://github.com/multiformats/js-multibase/issues/90,
+ * will be solved by https://github.com/jsdom/jsdom/issues/2524
+ *
+ * @jest-environment node
+ *
+ */
+
+import { parseIdentitySha1 } from "./urn";
+import * as ethers from "ethers";
+
+describe("parseIdentitySha1", () => {
+  function expectSuccess(urn: string, expectedRoot: string) {
+    const actual = parseIdentitySha1(urn);
+    expect(ethers.utils.hexlify(actual)).toEqual(expectedRoot);
+  }
+
+  function expectFailure(urn: string, expectedError: string) {
+    expect(() => parseIdentitySha1(urn)).toThrow(expectedError);
+  }
+
+  it("parses a valid identity URN", () => {
+    expectSuccess(
+      "rad:git:hnrkr5f1pdx7jysj5wkep9ci3pfaztfrgt7ay",
+      "0x4d964d1bfa90593ba290dfb2b969717894868f70"
+    );
```

Comment:
I’d prefer to inline the `expectSuccess` function since it is only used one but I’m also ok if you want to leave it like this. In general I really started to like this pattern of extracting the test body :sparkles:.

Reply:
I'd prefer keeping it separate. We're wasting just a few lines of code, but we're keeping the whole series of tests clear and consistent, we're also encouraging writing more success cases if they are needed.